Automattic/wp-calypso
# Earn Page: Add Launchpad button

## Proposed Changes

We recently introduce paid newsletters, which includes multiple links from Launchpad to specific Earn pages for connecting stripe and setting up a first newsletter plan. The issue is that once you click those links, it is difficult and not obvious for users to get back to Launchpad. 

This PR adds a new section to the top of Earn pages when a) Launchpad is enabled and b) site intent is newsletter. If neither (a) nor (b) are true, then Earn pages should load the same as always. 

See slack discussion here: 

**Screenshot of new appearance:**
<img width="1234" alt="earn-launchpad-2" src="https://github.com/Automattic/wp-calypso/assets/21228350/5bcb8977-b3e3-4075-8e6e-fe8f68bbbe83">


## Testing Instructions

**Prepare site to connect to Stripe**
    - You can test most of this without connecting to stripe. You may miss one screen (when you return from Stripe), but if the others work, that will work. So if time pressed, just test without. 
    - If you do want to connect to stripe, you'll want to add `define( 'USE_STORE_SANDBOX', true );` to wp-content/mu-plugins/0-sandbox.php in your sandbox. Then sandbox public-api.wordpress.com. You'll want to do this before you create your site. If you do this, when you arrive at Stripe, you'll see a note that you're in testing mode, and link to skip the connecting process. 

**Testing**
1. Check out this branch and run yarn and yarn start if needed. 
2. Start a newsletter site at http://calypso.localhost:3000/setup/newsletter/intro, select the option for a paid newsletter, and proceed to Launchpad. 
3. TEST: On Launchpad, click the option to `Setup payment method`. When you arrive on the Earn section, confirm you see the Launchpad nav section like the one in the screenshot on the page. Connect your site to stripe. When you return to the Earn area, again confirm you see the Launchpad nav section. 
4. TEST: Click the `Next Steps` button and confirm you are taken back to Launchpad. 
5. TEST: On Launchpad, click the option to `Create paid newsletter`. Create your plan. When the plan modal goes aways, confirm you see the Launchpad nav section. 
6. TEST: Click the `Next Steps` button and confirm you are taken back to Launchpad. 





*

## Pre-merge Checklist
